pinot-common/src/main/antlr4/org/apache/pinot/pql/parsers/PQL2.g4

```diff
@@ -77,6 +77,7 @@ predicate:
   | betweenClause                         # BetweenPredicate
   | isClause                              # IsPredicate
   | regexpLikeClause                      # RegexpLikePredicate
+  | matchesClause                         # MatchesPredicate  
```

Comment:
<nit> Would be good to keep this change to just object type changes and bring in the pql changes in a different PR</nit>